ext/sqlite3/database.c

```diff
@@ -12,6 +12,11 @@
 
 VALUE cSqlite3Database;
 
+// Function to check if two timespec structures are equal
+int timespec_equal(const struct timespec* ts1, const struct timespec* ts2) {
+    return (ts1->tv_sec == ts2->tv_sec) && (ts1->tv_nsec == ts2->tv_nsec);
+}
+
```

Comment:
```suggestion
```

Seems unused?

Reply:
Ah, yes. I had a usage and then refactored it away. Will drop. Thanks.